src/NServiceBus.Core/Pipeline/Outgoing/RoutingToDispatchConnector.cs

```diff
@@ -52,7 +40,34 @@ public override Task Invoke(IRoutingContext context, Func<IDispatchContext, Task
             return stage(this.CreateDispatchContext(operations, context));
         }
 
+        static void LogOutoingOperationsWhenDebugEnabled(TransportOperation[] operations)
+        {
+            if (!isDebugEnabled)
+            {
+                return;
+            }
+
+            StringBuilder sb = null;
+            foreach (var operation in operations)
+            {
+                sb = sb ?? new StringBuilder();
```

Comment:
Is there a chance `operations` can be empty? If not, then this optimization around avoiding creating the `StringBuilder` is unnecessary.